tools/validate/validateYaml.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+
+#
+# Validate that keys in yaml files are in the expected format:
+#   - camel case
+#   - acronyms are not capitalized (eg. clientId, not clientID; eventsUrl, not eventsURL)
+#
+
+import argparse
+import fnmatch
+import os
+import re
+import sys
+import yaml
+
+SCRIPT_DIRECTORY = os.path.dirname(os.path.realpath(__file__))
+ROOT_DIRECTORY = os.path.realpath(SCRIPT_DIRECTORY + "/../..")
+
+DESCRIPTION = 'Validate keys format in YAML files'
+
+parser = argparse.ArgumentParser(description=DESCRIPTION)
+parser.add_argument('-v', '--verbose', help="print verbose output",  action='store_true')
+args = parser.parse_args()
+
+def validate_yaml_key_format(key):
+  valid = True
+  # must be alphanumeric
+  if not re.match("^[a-z][a-zA-Z0-9_\-]*$", key):
+    valid = False
+  # doesn't contain capitalized acronyms
+  elif re.search("[A-Z]{2,}", key):
+    valid = False
+
+  if args.verbose:
+    print("key {}: {}".format(key, valid))
+
+  if not valid:
+    print("ERROR: invalid key {}".format(key), file=sys.stderr)
+  return valid
+
+def validate_yaml_keys(data):
+  valid = True
+  for k, v in data.items():
+    if not validate_yaml_key_format(k):
+      valid = False
+    if isinstance(v, dict):
+      if not validate_yaml_keys(v):
+        valid = False
+    elif isinstance(v, list):
+      for item in v:
+        if isinstance(item, dict):
+          if not validate_yaml_keys(item):
+            valid = False
+  return valid
+
+def find_and_validate_yaml_files(dir = ROOT_DIRECTORY):
+  valid = True
+  exclude_dirs = set(["dependency", "charts"])
```

Comment:
@ondrejtomcik I've skipped the whole `charts` folder, but I think stuff needs to be updated there as well. I'm not sure how the yaml files in charts/../templates work. For example:

```
webOAuthClient:
   clientID: {{ required "httpgateway.ui.webConfiguration.webOAuthClient.clientID or global.oauth.web.clientID is required" ( .ui.webConfiguration.webOAuthClient.clientID | default $.Values.global.oauth.web.clientID ) | quote }}
```

I think `webOAuthClient:` and `clientID` are just yaml fields that should be updated, but I'm not sure what to do with the stuff inside `{{ ... }}`

Reply:
The `templates` folder contains key: definitions for configmaps -> config.yaml of services. Here you need to change it to `clientId`. In the brackets you have the path to value, which is added there during templating, from custom user file or from https://github.com/plgd-dev/hub/blob/main/charts/plgd-hub/values.yaml (e.g. check line 27). If we change it, we introduce a breaking change. Let's not do it. Change only templates/..configmaps and their keys to match what service expects. But templating of helm chart, let's keep it as it is. There might be already user with his custom values file, setting clientID for the helm chart as a value.